daskms/experimental/zarr/__init__.py

```diff
@@ -375,12 +386,14 @@ def xds_from_zarr(store, columns=None, chunks=None, **kwargs):
 
     datasets = []
     numpy_vars = []
-
-    # NOTE(JSKenyon): Iterating over all the zarr groups/arrays is VERY
-    # expensive if the metadata has not been consolidated.
-    zc.consolidate_metadata(store.map)
     table_path = store.table if store.table else "MAIN"
-    table_group = zarr.open_consolidated(store.map)[table_path]
+
+    store_map = store.fs.get_mapper(f"{store.root}{store.fs.sep}{table_path}")
+
+    try:
+        table_group = zarr.open_consolidated(store_map)
```

Comment:
When reading, `xds_from_zarr` will attempt to read consolidated metadata, but fall back if it can't find it.